frontend/lib/ui/dashboard/artefact_dialog.dart

```diff
@@ -302,7 +304,73 @@ class _TestExecutionView extends StatelessWidget {
         ],
       ),
       expandButtonPosition: YaruExpandableButtonPosition.start,
-      child: const SizedBox.shrink(),
+      child: Padding(
+        padding: const EdgeInsets.only(left: Spacing.level4),
+        child: Column(
+          children: TestResultStatus.values
+              .map(
+                (status) => _TestResultsFilter(
+                  statusToFilterBy: status,
+                  testExecutionId: testExecution.id,
+                ),
+              )
+              .toList(),
+        ),
+      ),
+    );
+  }
+}
+
+class _TestResultsFilter extends ConsumerWidget {
```

Comment:
Not sure if Flutter has a rule around how Widgets should be organised in different files, but I see this one has grown to over 300 lines. Could we split this method in a new file, maybe `test_case_results.dart` or even the whole TestExecution view in `test_executions.dart`, might be even more helpful when we start adding the signoff stuff. What do you think about this?

Reply:
Completely agree, I've been neglecting this problem for too long. Will separate them into files